homeassistant/components/history_stats/sensor.py

```diff
@@ -115,7 +149,11 @@ def __init__(
         self._end = end
         self._type = sensor_type
         self._name = name
-        self._unit_of_measurement = UNITS[sensor_type]
+        self._precision = precision
+        self._time_unit = time_unit
+        self._unit_of_measurement = (
+            UNITS[sensor_type] if not sensor_type == CONF_TYPE_TIME else time_unit
```

Comment:
I feel like a quick comment wouldn't hurt here to explain what's going on.

Reply:
Sounds reasonable.  I'll add a comment to explain the handling of units.